stdlib/array.ncl

```diff
@@ -116,12 +117,62 @@
 
     fold_left : forall a b. (a -> b -> a) -> a -> Array b -> a
       | doc m%"
-        Fold a function over an array.
+        Fold a function over an array. Folds serve a similar purpose to loops or
+        iterator in a functional language like Nickel. `fold_left` iterates over
+        an array, by repeatedly applying a function over each element, threading
+        an additional arbitrary state (the accumulator, of type `a` in the
+        signature).
+
         `fold_left f init [x1, x2, ..., xn]` results in `f (... (f (f init x1) x2) ...) xn`.
 
         This function is strict in the intermediate accumulator.
 
-        For example:
+        # Left vs right
+
+        Folds come in two variants, left and right. How to decide which one to
+        use?
+
+        - If the folded function isn't associative (such as subtraction), then
+          each variant will give a different result. The choice is dictacted by
+          which one you need. For example:
+
+          ```nickel
+          array.fold_right (-) 0 [1, 2, 3, 4]
+           => -2
+          array.fold_left (-) 0 [1, 2, 3, 4]
+           => -10
+          ```
+        - If the folded function is associative, both `fold_right` and
+          `fold_left` return the same result. In that case, **`fold_left` is
+          generally preferred**, because it forces the evaluation of the
+          intermediate results resulting in less memory consumption and overall
+          better performances (outside of pathological cases).
```

Comment:
I have no strong opinion on this one, but isn't "performances" as well? In that overall, it should lead to better performances, all considered?

Reply:
I don't think "performance" is a countable noun in English. At least I can't think of a situation where "performances" doesn't sound weird to me.

Reply:
It seems performance can be countable when used with this meaning (https://english.stackexchange.com/questions/367488/plural-of-performance). Still, it looks less common and natural than using the uncountable version. It might very well be my own deformation, as it's more common to use the pural "les performances" than the singular "la performance" in French, in this situation.